### PR TITLE
feat: Onboarding connect api token generation

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

When Clicking generate token or token exists for a given env you get token anatomy
Screenshot 2024-09-03 at 10 52 02

On mobile we move extra information and only show the token
Screenshot 2024-09-03 at 10 52 27

Details:
* this step will be moved after SDK selection because we need token type (client vs frontend)
* we will hardcode some token name
* about 80 LOC in this PR is extract existing inline components into separate components for readability

### CODE REVIEW

File: frontend/src/component/onboarding/ConnectSDKDialog.tsx [hunk 1]
```diff
@@ -185,49 +428,8 @@ export const ConnectSDKDialog = ({
                         </NextStepSectionSpacedContainer>
                     </NextStepSection>
                 </APIKeyGeneration>
-                <ConceptsDefinitions>
```
Comment (kwasniew) [inline frontend/src/component/onboarding/ConnectSDKDialog.tsx:188]: extracted to separate component for readability

File: frontend/src/component/onboarding/ConnectSDKDialog.tsx [hunk 2]
```diff
@@ -106,22 +114,252 @@ const NextStepSectionSpacedContainer = styled('div')(({ theme }) => ({
...
+                            </SecretExplanationDescription>
+                        </ArcherElement>
+                    </TokenExplanationBox>
+                ) : null}
+            </SecretExplanation>
+        </ArcherContainer>
+    );
+};
+
+const ChooseEnvironment = ({
```
Comment (kwasniew) [inline frontend/src/component/onboarding/ConnectSDKDialog.tsx:284]: extracted into component for readability

File: frontend/src/component/onboarding/ConnectSDKDialog.tsx [hunk 3]
```diff
@@ -106,22 +114,252 @@ const NextStepSectionSpacedContainer = styled('div')(({ theme }) => ({
...
+            project,
+            environment,
+            secret,
+        };
+    }
+
+    return null;
+}
+
+const ConceptsDefinitions = () => (
```
Comment (kwasniew) [inline frontend/src/component/onboarding/ConnectSDKDialog.tsx:154]: extracted for readability

File: frontend/src/component/onboarding/ConnectSDKDialog.tsx [hunk 4]
```diff
@@ -59,7 +67,7 @@ export const APIKeyGeneration = styled('div')(({ theme }) => ({
     backgroundColor: theme.palette.background.paper,
     display: 'flex',
     flexDirection: 'column',
-    flexBasis: '70%',
+    flex: 1,
```
Comment (kwasniew) [inline frontend/src/component/onboarding/ConnectSDKDialog.tsx:70]: take whole dialog in mobile view

File: frontend/src/component/onboarding/ConnectSDKDialog.tsx [hunk 5]
```diff
@@ -106,22 +114,252 @@ const NextStepSectionSpacedContainer = styled('div')(({ theme }) => ({
...
+const SecretExplanationDescription = styled('div')(({ theme }) => ({
+    backgroundColor: theme.palette.primary.contrastText,
+    borderRadius: theme.shape.borderRadius,
+    padding: theme.spacing(2),
+    flex: 1,
+    color: theme.palette.text.secondary,
+    fontSize: theme.fontSizes.smallBody,
+}));
+
+function splitToken(
```
Comment (sjaanus) [inline frontend/src/component/onboarding/ConnectSDKDialog.tsx:135]: Can this be const splitToken, and can we have test for it?

**Review by sjaanus — APPROVED** [review]
Approved with suggestion

File: frontend/src/component/onboarding/ConnectSDKDialog.tsx (hunk 5 above)
Comment (kwasniew) [reply frontend/src/component/onboarding/ConnectSDKDialog.tsx:135]: yup, will do

**Pull request merged**